ReactWindows/ReactNative.Shared/Views/ViewManagersPropertyCache.cs

```diff
@@ -32,7 +33,8 @@ static class ViewManagersPropertyCache
                 }
             }
 
-            s_settersCache.Add(type, settersImpl);
+            // Make sure it works on concurrent accesses that both created a value for same key
+            s_settersCache.AddOrUpdate(type, settersImpl, (key, value) => value);
```

Comment:
I faintly remember that to make a dictionary thread safe there was somethin in addition to making a concurrentdictionary and using AddOrUpdate pattern (I can't recall exactly what). You can ping me offline if you decide to keep the concurrentdictionary instead of going with what Eric suggests and make separate instances...

Reply:
Talked to Radu offline and resolved.

---
In reply to: [166082566](https://github.com/Microsoft/react-native-windows/pull/1615#discussion_r166082566) [](ancestors = 166082566)